mmhuman3d/core/visualization/visualize_smpl.py

```diff
@@ -802,41 +811,58 @@ def render_smpl(
         raise ValueError('Please choose the right render_choice.')
 
     # body part colorful visualization should use flat shader to be sharper.
-    if isinstance(palette, str):
-        if (palette == 'segmentation') and ('silhouette'
-                                            not in render_choice.lower()):
-            render_param_dict['shader_type'] = 'flat'
-
-        palette = [palette] * num_person
-    elif isinstance(palette, np.ndarray):
-        palette = torch.Tensor(palette)
-        palette = palette.view(-1, 3)
-        if palette.shape[0] != num_person:
-            _times = num_person // palette.shape[0]
-            palette = palette.repeat(_times, 1)[:num_person]
-            if palette.shape[0] == 1:
-                print(f'Same color for all the {num_person} people')
-            else:
-                print('Repeat palette for multi-person.')
-    else:
-        raise ValueError('Wrong input palette type. '
-                         'Palette should be tensor, array or list of strs')
-    colors = _prepare_colors(palette, render_choice, num_person, num_verts,
-                             model_type)
+    if texture_image is None:
+        if isinstance(palette, str):
+            palette = [palette] * num_person
+        elif isinstance(palette, np.ndarray):
+            palette = torch.Tensor(palette)
+            palette = palette.view(-1, 3)
+            if palette.shape[0] != num_person:
+                _times = num_person // palette.shape[0]
+                palette = palette.repeat(_times, 1)[:num_person]
+                if palette.shape[0] == 1:
+                    print(f'Same color for all the {num_person} people')
+                else:
+                    print('Repeat palette for multi-person.')
+        else:
+            raise ValueError('Wrong input palette type. '
+                             'Palette should be tensor, array or list of strs')
+        colors_all = _prepare_colors(palette, render_choice, num_person,
+                                     num_verts, model_type)
+        colors_all = colors_all.view(-1, num_person * num_verts, 3)
+
```

Comment:
As `verts` argument of ParametricMeshes is required to be in shape [N, V, 3]  or a list, here we should follow the docstring, even though it doesn't cause an error.
`vertices = vertices.view(num_frames, -1, 3)`

Reply:
ok